internal/testutils/parse_url.go

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package testutils
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// MustParseURL uses RFC 3986 semantics to parse given target into a
+// `URL` struct for `resolver.Target.URL`. The leading "/" are stripped
+// to avoid breaking existing resolver implementations.
+// See resolver.Target.ToEndpoint() for more information.
+//
+// rawTarget represents either URL, IPv4, or IPv6 string.
+func MustParseURL(rawTarget string) *url.URL {
+	u, err := url.Parse(rawTarget)
+	if err != nil {
+		panic(fmt.Sprintf("Error parsing target(%s): %v", rawTarget, err))
+	}
```

Comment:
Is there a problem in returning `u` instead?

Even in `grpc.parseTarget()`, we store the parsed url as is in `resolver.Target.URL`. So, I don't understand why we should return a new `url.URL` here that we manually populate.

Reply:
I fixed the function to return `u` and revised the documentation. Thanks for your feedback!

> Even in grpc.parseTarget(), we store the parsed url as is in resolver.Target.URL

I overlooked this detail and mistakenly thought it was important to handle the leading "/" here too.

If this was uncaught, would that potentially result in tests failing to detect issues with `resolver.Target.ToEndpoint` since `MustParseURL` stripped the leading "/" outright?

Reply:
Done. Thanks again for all your feedback. I learned a lot.

FYI - I also rebased to capture latest changes from `master` and squashed my commits to help clean up.